balboa-kafka/src/main/scala/com/socrata/balboa/kafka/consumer/Consumer.scala

```diff
@@ -0,0 +1,74 @@
+package com.socrata.balboa.kafka.consumer
+
+import java.io.IOException
+
+import com.socrata.balboa.metrics.data.DataStore
+import com.socrata.balboa.metrics.impl.JsonMessage
+import kafka.consumer.{ConsumerIterator, KafkaStream}
+
+/**
+ * Created by Michael Hotan on 2/18/15.
+ *
+ * Balboa Kafka Consumer
+ */
+sealed abstract class BalboaConsumer(stream: KafkaStream[Array[Byte], Array[Byte]]) extends Runnable {
+
+  override def run(): Unit = {
+    val it: ConsumerIterator[Array[Byte], Array[Byte]] = stream.iterator()
+    while (it.hasNext())
+      onMessage(it.next().message())
+  }
+
+  /**
+   * Notification of message receipt.
+   *
+   * @param message Received Kafka Message
+   */
+  def onMessage(message: Array[Byte]): Unit
+
+  /**
+   * Converts an array of bytes to a String where each character is the
+   *
+   * @param bytes The bytes to convert.
+   * @return The String representation of the Byte Array.
+   */
+  def bytesToString(bytes: Array[Byte]): String = {
+    new String(bytes.map(_.toChar))
+  }
+
+}
+case class TestConsumer(stream: KafkaStream[Array[Byte], Array[Byte]]) extends BalboaConsumer(stream) {
```

Comment:
So I don't see an actual integration test that uses the test consumer yet. It seems like we want to start building the unit tests right away.